packages/block-library/src/embed/edit.js

```diff
@@ -33,181 +35,215 @@ function getResponsiveHelp( checked ) {
 
-		// Even though we set attributes that get derived from the preview,
-		// we don't access them directly because for the initial render,
-		// the `setAttributes` call will not have taken effect. If we're
-		// rendering responsive content, setting the responsive classes
-		// after the preview has been rendered can result in unwanted
-		// clipping or scrollbars. The `getAttributesFromPreview` function
-		// that `getMergedAttributes` uses is memoized so that we're not
-		// calculating them on every render.
-		const previewAttributes = getMergedAttributes(
-			props,
-			title,
-			responsive
-		);
-		const { caption, type, allowResponsive } = previewAttributes;
-		const className = classnames(
-			previewAttributes.className,
-			props.className
-		);
+	if ( fetching ) {
+		return <EmbedLoading />;
+	}
+
+	// translators: %s: type of embed e.g: "YouTube", "Twitter", etc. "Embed" is used when no specific type exists
+	const label = sprintf( __( '%s URL' ), title );
 
+	// No preview, or we can't embed the current URL, or we've clicked the edit button.
+	const showEmbedPlacholder = ! preview || cannotEmbed || isEditingURL;
+	if ( showEmbedPlacholder ) {
 		return (
-			<>
-				<EmbedControls
-					showEditButton={ preview && ! cannotEmbed }
-					themeSupportsResponsive={ themeSupportsResponsive }
-					blockSupportsResponsive={ responsive }
-					allowResponsive={ allowResponsive }
-					getResponsiveHelp={ getResponsiveHelp }
-					toggleResponsive={ toggleResponsive }
-					switchBackToURLInput={ () => setIsEditingURL( true ) }
-				/>
-				<EmbedPreview
-					preview={ preview }
-					previewable={ previewable }
-					className={ className }
-					url={ url }
-					type={ type }
-					caption={ caption }
-					onCaptionChange={ ( value ) =>
-						setAttributes( { caption: value } )
+			<EmbedPlaceholder
+				icon={ icon }
+				label={ label }
+				onSubmit={ ( event ) => {
+					if ( event ) {
+						event.preventDefault();
 					}
-					isSelected={ isSelected }
-					icon={ icon }
-					label={ label }
-					insertBlocksAfter={ insertBlocksAfter }
-				/>
-			</>
+
+					setIsEditingURL( false );
+					setAttributes( { url } );
+				} }
+				value={ url }
+				cannotEmbed={ cannotEmbed }
+				onChange={ ( event ) => setURL( event.target.value ) }
+				fallback={ () => fallback( url, onReplace ) }
+				tryAgain={ () => {
+					invalidateResolution( 'core', 'getEmbedPreview', [ url ] );
+				} }
+			/>
 		);
-	};
-}
+	}
+
+	// Even though we set attributes that get derived from the preview,
+	// we don't access them directly because for the initial render,
+	// the `setAttributes` call will not have taken effect. If we're
+	// rendering responsive content, setting the responsive classes
+	// after the preview has been rendered can result in unwanted
+	// clipping or scrollbars. The `getAttributesFromPreview` function
+	// that `getMergedAttributes` uses is memoized so that we're not
+	// calculating them on every render.
```

Comment:
// Do we need this? Doesn't `setAttributes` cause a rerender, so the new classes would be applied?